app/models/linked_profile.rb

```diff
@@ -31,6 +31,35 @@ class LinkedProfile < ApplicationRecord
   belongs_to :user, required: true
   belongs_to :linked_site, required: true
 
+  attr_encrypted :token, key: Base64.encode64(ENV['ATTR_ENCRYPT_KEY'])
+
   validates_presence_of :url, if: :private?
   validates_presence_of :external_user_id
+  validate :verify_mal_credentials
+
+  def verify_mal_credentials
+    # Check to make sure username/password is valid
+    host = MyAnimeListSyncService::ATARASHII_API_HOST
+    response = Typhoeus::Request.new(
+      "#{host}account/verify_credentials",
+      method: :get,
+      userpwd: "#{external_user_id}:#{token}"
+    ).run
+
+    unless response.code == 200
+      errors.add(:token, "#{response.code}: #{response.body}")
+    end
+
+    true
+  end
+
+  def sync_to_mal?
+    User.find(user_id).linked_profiles.where(
+      linked_site: LinkedSite.find_by(name: 'myanimelist')
+    ).present?
+  end
```

Comment:
This seems a bit... roundabout

Couldn't you just do `linked_site.name == 'myanimelist' && token.present?`